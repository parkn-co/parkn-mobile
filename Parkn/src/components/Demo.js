import React, {Component} from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View
} from 'react-native';

function Demo(props) {
  return (
    <View style={styles.container}>
      <Text style={styles.welcome}>
        Welcome to React Native!
      </Text>
      <Text style={styles.instructions}>
        To get started, edit index.ios.js
      </Text>
      <Text style={styles.instructions}>
        Press Cmd+R to reload,{'\n'}
        Cmd+D or shake for dev menu
      </Text>
    </View>
  );
}

Demo.route = {
  title: 'Hello!',
  nav: {
    left: (route, navigator, index, navState) => ({
      onPress: (route, navigator, index, navState) => {},
      text: (route, navigator, index, navState) => 'Sup',
    }),
    title: (route, navigator, index, navState) => <Text>{'Hello'}</Text>,
    right: (route, navigator, index, navState) => ({
      onPress: (route, navigator, index, navState) => {},
      text: (route, navigator, index, navState) => 'Yo',
    }),
  }
};

export default Demo;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
