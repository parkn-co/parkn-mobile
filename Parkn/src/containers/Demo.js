
// @flow
import type {User} from 'flow-declarations/user';

import React, {Component} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {signOut} from 'actions/authentication';
import Demo from 'components/Demo';

type Props = {
  handleSignOut: () => void,
  user: User,
};

const DemoContainer = ({handleSignOut, user}: Props): React.Element<*> => (
  <Demo handleSignOut={handleSignOut} user={user} />
);

const mapDispatchToProps = (dispatch: Function): any => {
  return bindActionCreators({
    handleSignOut: signOut,
  }, dispatch);
};

export default connect(null, mapDispatchToProps)(DemoContainer);

