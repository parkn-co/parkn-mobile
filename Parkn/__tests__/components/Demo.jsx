/* eslint-env jest */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Demo from '../../src/components/Demo';

describe('Demo', () => {
  it('renders correctly using snapshots', () => {
    const user = {
      email: 'jared@example.com',
      firstName: 'jared',
      lastName: 'ramirez',
    };
    expect(renderer.create(
      <Demo handleSignOut={jest.fn()} user={user} />
    )).toMatchSnapshot();
  });
});
