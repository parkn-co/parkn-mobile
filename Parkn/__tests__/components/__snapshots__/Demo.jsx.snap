exports[`Demo renders correctly using snapshots 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }>
    Welcome to Parkn, jared!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }>
    Full Name: jared ramirez
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }>
    Email: jared@example.com
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#333",
        "borderWidth": 1,
        "margin": 20,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Sign Out
    </Text>
  </View>
</View>
`;
